osu.Game.Tests/Visual/Gameplay/TestScenePauseWhenInactive.cs

```diff
@@ -33,10 +33,53 @@ public void TestDoesntPauseDuringIntro()
 
             AddStep("resume player", () => Player.GameplayClockContainer.Start());
             AddAssert("ensure not paused", () => !Player.GameplayClockContainer.IsPaused.Value);
+
+            AddStep("progress time to gameplay", () => Player.GameplayClockContainer.Seek(Player.DrawableRuleset.GameplayStartTime));
+            AddUntilStep("wait for pause", () => Player.GameplayClockContainer.IsPaused.Value);
+        }
+
+        /// <summary>
+        /// Tests that if a pause from focus lose is performed while in pause cooldown,
+        /// the player will still pause after the cooldown is finished.
+        /// </summary>
+        [Test]
+        public void TestPauseWhileInCooldown()
+        {
+            AddStep("resume player", () => Player.GameplayClockContainer.Start());
+            AddStep("skip to gameplay", () => Player.GameplayClockContainer.Seek(Player.DrawableRuleset.GameplayStartTime));
+
+            AddStep("set inactive", () => ((Bindable<bool>)host.IsActive).Value = false);
+            AddUntilStep("wait for pause", () => Player.GameplayClockContainer.IsPaused.Value);
+
+            AddStep("set active", () => ((Bindable<bool>)host.IsActive).Value = true);
+
+            AddStep("resume player", () => Player.Resume());
+            AddStep("click resume overlay", () =>
+            {
+                InputManager.MoveMouseTo(this.ChildrenOfType<OsuResumeOverlay.OsuClickToResumeCursor>().Single());
+                InputManager.Click(MouseButton.Left);
+            });
```

Comment:
Why not do what other tests do and move the mouse out of the playfields so you don't need to worry about this stuff? It's out of scope of this test case.

Reply:
so that's how it's done, haven't searched properly..